android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt

```diff
@@ -411,6 +413,8 @@ open class GestureHandler<ConcreteGestureHandlerT : GestureHandler<ConcreteGestu
   protected open fun onStateChange(newState: Int, previousState: Int) {}
   protected open fun onReset() {}
   protected open fun onCancel() {}
+  protected open fun beforeActivation() {}
```

Comment:
This seem weirdly specific to activation only. We already have a bunch of patterns around placing state-change logic. E.g., we have methods like onCancel or onStateChange. I wonder if we could use one of these? Alternatively we can consider changing functions like activate to not be final and this way allow them to be overriden which would seem like a cleaner approach, no?

Reply:
I don't think it would be possible using one of the existing methods (besides adding the relevant code alongside every `activate` call), but making `activate` to not be final seems like a much better approach. I updated it in https://github.com/software-mansion/react-native-gesture-handler/pull/1670/commits/7db36d5dd2ba5fba4c1fca9f5cdbcc0da58e5705.